cwatm/hydrological_modules/water_demand/water_demand.py

```diff
@@ -931,7 +1056,302 @@ def dynamic(self):
+    
+                        self.var.abstractedLakeReservoirM3C += act_bigLakeAbst_wwtC.copy() * self.var.MtoM3C
+                        self.var.lakeResStorage = globals.inZero.copy()
+                        np.put(self.var.lakeResStorage, self.var.decompress_LR, self.var.lakeResStorageC)
+                        bigLakesFactor_wwtC = divideValues(act_bigLakeAbst_wwtC, remainNeedBig_wwtC)
+                        # and back to the big array
+                        bigLakesFactor_wwt = globals.inZero.copy()
+                        np.put(bigLakesFactor_wwt, self.var.decompress_LR, bigLakesFactor_wwtC)
+
+                        # bigLakesFactorAllaroundlake = npareamaximum(bigLakesFactor, self.var.waterBodyID)
+                        bigLakesFactorAllaroundlake_wwt = npareamaximum(bigLakesFactor_wwt, self.var.waterBodyBuffer_wwt)
+                        #print(np.compress(self.var.compress_LR, bigLakesFactorAllaroundlake_wwt))
+                        # abstraction from big lakes is partioned to the users around the lake
+                        self.var.act_bigLakeResAbst_wwt = remainNeed * bigLakesFactorAllaroundlake_wwt
+                        # remaining need is used from lakes and reservoirs without water use restrictions
+                        #remainNeed0 = remainNeed * (1 - bigLakesFactorAllaroundlake_wwt)
+                       
+                        
+                        # allocate water demand in case sectorSourceAbstractionFractions = True
+                        if self.var.sectorSourceAbstractionFractions:
+                            self.var.wwt_Domestic = np.minimum(self.var.act_bigLakeResAbst_wwt,
+                                                           pot_wwt_Domestic)
+                            self.var.wwt_Livestock = np.minimum(self.var.act_bigLakeResAbst_wwt - self.var.wwt_Domestic,
+                                                                pot_wwt_Livestock)
+                            self.var.wwt_Industry = np.minimum(
+                                self.var.act_bigLakeResAbst_wwt - self.var.wwt_Domestic - self.var.wwt_Livestock,
+                                pot_wwt_Industry)
+                            
+                            self.var.wwt_Irrigation = np.minimum(
+                                self.var.act_bigLakeResAbst_wwt - self.var.wwt_Domestic - self.var.wwt_Livestock - self.var.wwt_Industry,
+                                pot_wwt_Irrigation)
+                    else:
+                        ## opt 2:  command areas are used
+                        #self.var.abstractedLakeReservoirM3C = np.compress(self.var.compress_LR, globals.inZero.copy())
+                        self.var.act_ResAbst_wwt = globals.inZero.copy()
+                        self.var.act_bigLakeResAbst_wwt = globals.inZero.copy()
+                        
+                        if self.var.modflow and 'Water_conveyance_efficiency' in binding:
```

Comment:
This is now only activated if wastewater is activated, to be moved outside

Reply:
Correct. I have now added it again in line 1697 some it applies also to the non-wastewater implementation of command areas.

Reply:
Yep.  Now in line 461.